waku/v2/node/rest/relay/relay_api.nim

```diff
@@ -0,0 +1,200 @@
+
+    let reqBodyData = contentBody.get().data
+    let reqResult = decodeFromJsonBytes(RelayPostSubscriptionsRequest, reqBodyData)
+    if reqResult.isErr():
+      return RestApiResponse.badRequest()
+
+    let req: RelayPostSubscriptionsRequest = reqResult.get()
+
+    for topic in req:
+      if topicCache.isSubscribed(string(topic)):
+        # Only subscribe to topics for which we have no subscribed topic handlers yet
+        continue
+
+      topicCache.subscribe(string(topic))
+      node.subscribe(string(topic), topicCache.messageHandler())
+
+    return RestApiResponse.ok()
+
+
+proc installRelayDeleteSubscriptionsV1Handler*(router: var RestRouter, node: WakuNode, topicCache: TopicCache) =
+  router.api(MethodDelete, ROUTE_RELAY_SUBSCRIPTIONSV1) do (contentBody: Option[ContentBody]) -> RestApiResponse:
+    # ## Subscribes a node to a list of PubSub topics
+    # debug "delete_waku_v2_relay_v1_subscriptions"
+
+    # Check the request body
+    if contentBody.isNone():
+      return RestApiResponse.badRequest()
+    
+    let reqBodyContentType = MediaType.init(contentBody.get().contentType)
+    if reqBodyContentType != MIMETYPE_JSON:
+      return RestApiResponse.badRequest()
+
+    let reqBodyData = contentBody.get().data
+    let reqResult = decodeFromJsonBytes(RelayDeleteSubscriptionsRequest, reqBodyData)
+    if reqResult.isErr():
+      return RestApiResponse.badRequest()
+
+    let req: RelayDeleteSubscriptionsRequest = reqResult.get()
+
+    # Unsubscribe all handlers from requested topics
+    for topic in req:
+      node.unsubscribeAll(string(topic))
+      topicCache.unsubscribe(string(topic))
+
+    # Successfully unsubscribed from all requested topics
+    return RestApiResponse.ok() 
+
+
+const ROUTE_RELAY_MESSAGESV1* = "/relay/v1/messages/{topic}"
+
+proc installRelayGetMessagesV1Handler*(router: var RestRouter, node: WakuNode, topicCache: TopicCache) =
+  router.api(MethodGet, ROUTE_RELAY_MESSAGESV1) do (topic: string) -> RestApiResponse: 
+    # ## Returns all WakuMessages received on a PubSub topic since the
+    # ## last time this method was called
+    # ## TODO: ability to specify a return message limit
+    # debug "get_waku_v2_relay_v1_messages", topic=topic
+
+    if topic.isErr():
+      return RestApiResponse.badRequest()
+    let pubSubTopic = topic.get()
+
+    let messages = topicCache.getMessages(pubSubTopic, clear=true)
+    if messages.isErr():
+      debug "Not subscribed to topic", topic=pubSubTopic
+      return RestApiResponse.notFound() 
+    
+    let data = RelayGetMessagesResponse(messages.get().map(toRelayWakuMessage))
+    let resp = RestApiResponse.jsonResponse(data, status=Http200)
+    if resp.isErr():
+      debug "An error ocurred while building the json respose", error=resp.error()
+      return RestApiResponse.internalServerError()
+
+    return resp.get()
+ 
+proc installRelayPostMessagesV1Handler*(router: var RestRouter, node: WakuNode, topicCache: TopicCache) =
```

Comment:
It's just a `ref object`, but is the `topicCache` necessary when installing a post message handler?

Reply:
You are right. Actually, I left it there to see if you were paying attention to my PR 😜